components/_helpers.js

```diff
@@ -1,19 +1,21 @@
 const modSep = '_';
 
-function getClassNamesWithMods(baseClass = '', mods = {}) {
+function getClassNamesWithMods(baseClass = '', ...modsSet) {
   let classes = [].concat(baseClass);
   baseClass = classes[0];
 
-  if (mods instanceof Array) {
-    classes = classes
-      .concat(mods.filter(Boolean).map(mode => `${baseClass}${modSep}${mode}`));
-  } else {
-    Object.keys(mods).forEach((mode) => {
-      if (mods[mode]) {
-        classes.push(`${baseClass}${modSep}${mode}`);
-      }
-    });
-  }
+  modsSet.forEach((mods) => {
```

Comment:
Seriously? micro optimization? better to write simple, more readable code, instead of try to micro optimize. I'm not sure if it will works more slowly then a lot of constructions like this: `if () {}`. 
BTW, at this moment all components has only one argument.
from my point of view, it's better way.
```
const className = getClassNamesWithMods('ui-badge-badge', this.props.mods, badgeMods);
```
*VS*
```
const mods = props.mods ? props.mods.slice() : [];
if (disabled) {
    mods.push(`disabled_true`);
}
```
In additional, you no need to create new `Array` in `render` method, because `getClassNamesWithMods` didn't change mobs from props, just return `string`